tests/posix/common/src/pthread.c

```diff
@@ -465,6 +465,40 @@ ZTEST(pthread, test_pthread_cleanup)
 	zassert_ok(pthread_join(th, NULL));
 }
 
+static bool testcancel_failed = true;
+
+static void *test_pthread_cancel_fn(void *arg)
+{
+	zassert_ok(pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, NULL));
+
+	/* this should be ignored */
+	pthread_testcancel();
+
+	/* this will mark it pending */
+	pthread_cancel(pthread_self());
```

Comment:
```suggestion
	zassert_ok(pthread_cancel(pthread_self()));
```

https://github.com/zephyrproject-rtos/zephyr/pull/67020/files#r1474142105